systemtest/src/test/java/io/strimzi/systemtest/log/LoggingChangeST.java

```diff
@@ -174,6 +183,179 @@ void testJSONFormatLogging() {
         assertThat(StUtils.checkLogForJSONFormat(eoPods, "user-operator"), is(true));
     }
 
+    @Test
+    @SuppressWarnings({"checkstyle:MethodLength"})
+    void testDynamicallySetEOloggingLevels() throws InterruptedException {
+        InlineLogging ilOff = new InlineLogging();
+        ilOff.setLoggers(Collections.singletonMap("rootLogger.level", "OFF"));
+
+        KafkaResource.kafkaPersistent(CLUSTER_NAME, 1, 1)
+                .editSpec()
+                    .editEntityOperator()
+                        .editTopicOperator()
+                            .withInlineLogging(ilOff)
+                        .endTopicOperator()
+                        .editUserOperator()
+                            .withInlineLogging(ilOff)
+                        .endUserOperator()
+                    .endEntityOperator()
+                .endSpec()
+                .done();
+
+        String eoDeploymentName = KafkaResources.entityOperatorDeploymentName(CLUSTER_NAME);
+        Map<String, String> eoPods = DeploymentUtils.depSnapshot(eoDeploymentName);
+
+        final String eoPodName = eoPods.keySet().iterator().next();
+
+        LOGGER.info("Changing rootLogger level to INFO with inline logging");
+        InlineLogging ilInfo = new InlineLogging();
+        ilInfo.setLoggers(Collections.singletonMap("rootLogger.level", "INFO"));
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            k.getSpec().getEntityOperator().getTopicOperator().setLogging(ilInfo);
+            k.getSpec().getEntityOperator().getUserOperator().setLogging(ilInfo);
+        });
+
+        LOGGER.info("The EO shouldn't roll - verifying pod stability");
+        DeploymentUtils.waitForNoRollingUpdate(eoDeploymentName, eoPods);
+
+        LOGGER.info("Waiting for log4j2.properties will contain desired settings");
+        TestUtils.waitFor("Logger change", Constants.GLOBAL_POLL_INTERVAL, Constants.GLOBAL_TIMEOUT,
+            () -> cmdKubeClient().execInPodContainer(eoPodName, "topic-operator", "cat", "/opt/topic-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=INFO")
+                        && cmdKubeClient().execInPodContainer(eoPodName, "user-operator", "cat", "/opt/user-operator/custom-config/log4j2.properties").out().contains("rootLogger.level=INFO")
+        );
+
+        LOGGER.info("Waiting {} ms for INFO log will appear", LOGGING_RELOADING_INTERVAL);
+        //wait some time if TO and UO will log something
+        Thread.sleep(LOGGING_RELOADING_INTERVAL);
```

Comment:
Should you wait double the interval? With reloading every 30 seconds, this might basically hit at the same time without any log message, or?